### ISSUE TITLE
Feature request: bundle wal2json

### ISSUE THREAD

**Issue by benjie** [PostgresApp/PostgresApp#496]
Great app, thanks for making PostgreSQL so easy to install!

I'd like to request that you bundle the `wal2json` output plugin with PostgresApp. It only takes a couple seconds to build, but unfortunately because PostgresApp is built with a specific version of the MacSDKs, users who do not have that SDK installed cannot build against it. (Note many cloud providers such as [Amazon RDS bundle wal2json](https://aws.amazon.com/about-aws/whats-new/2018/11/amazon-rds-postgresql-105-govcloud/).)

For example, the `pg_config` in this user's Postgres.app indicates `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk`, but this user only has `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk` available:

```
+ rm -Rf wal2json
+ git clone https://github.com/eulerto/wal2json.git
Cloning into 'wal2json'...
remote: Enumerating objects: 43, done.
remote: Counting objects: 100% (43/43), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 492 (delta 24), reused 29 (delta 19), pack-reused 449
Receiving objects: 100% (492/492), 203.11 KiB | 3.44 MiB/s, done.
Resolving deltas: 100% (343/343), done.
+ cd wal2json
+ USE_PGXS=1
+ make
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument  -mmacosx-version-min=10.7  -I. -I./ -I/Applications/Postgres.app/Contents/Versions/10/include/postgresql/server -I/Applications/Postgres.app/Contents/Versions/10/include/postgresql/internal -I/Applications/Postgres.app/Contents/Versions/10/share/icu -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk  -I/Applications/Postgres.app/Contents/Versions/10/include/libxml2  -I/Applications/Postgres.app/Contents/Versions/10/include  -c -o wal2json.o wal2json.c
clang: warning: no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk' [-Wmissing-sysroot]
In file included from wal2json.c:13:
In file included from /Applications/Postgres.app/Contents/Versions/10/include/postgresql/server/postgres.h:47:
/Applications/Postgres.app/Contents/Versions/10/include/postgresql/server/c.h:81:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
make: *** [wal2json.o] Error 1
```

```
$ ls /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer//SDKs/
MacOSX.sdk      MacOSX10.14.sdk
```

Installing wal2json is straight-forward (and fast!):

```bash
PATH="/path/to/pg/bin:$PATH"
git clone https://github.com/eulerto/wal2json.git
cd wal2json
USE_PGXS=1 make
USE_PGXS=1 make install
```

**Comment by jakob** [comment]
Holy shit, they backported the PG_SYSROOT / -isysroot CPPFLAGS !#$##%!&*. I forgot all about that.

This is annoying. I'll need to make new builds of Postgres.app or people won't be able to build extensions.

Building extensions should work with version PostgreSQL 11 -- I added `PG_SYSROOT=no-sysroot` to avoid exactly this issue. As soon as I have time I'll make new builds of PostgreSQL 9.4 - 10.

**Comment by jakob** [comment]
Just a quick update on this: the latest release of Postgres.app (2.2.3) now builds all versions with `PG_SYSROOT=no-sysroot`. This means that building extensions against PostgreSQL 10 and earlier should work again as usual!

Please let me know if you run into any further trouble building extensions against Postgres.app.

**Comment by benjie** [comment]
Thanks @jakob; will let you know if I hear of any issues 👍

**Comment by n1k1taM** [comment]
> Just a quick update on this: the latest release of Postgres.app (2.2.3) now builds all versions with `PG_SYSROOT=no-sysroot`. This means that building extensions against PostgreSQL 10 and earlier should work again as usual!
> 
> Please let me know if you run into any further trouble building extensions against Postgres.app.

@jakob  Is anybody going to fix the same issues for PostgreSQL 9.4 - 10? We have the same issue `https://github.com/eulerto/wal2json/issues/117`

**Comment by jakob** [comment]
@n1k1taM The `PG_SYSROOT` issue should be fixed for all versions of Postgres.app.

But I don't really have any idea what you mean with "We have the same issue". Could you please open a new issue and explain in detail:

 - What are you trying to do?
- What version of macOS / Postgres.app are you using?
- What error message are you seeing?
